lms/djangoapps/courseware/module_render.py

```diff
@@ -991,7 +1019,13 @@ def xblock_view(request, course_id, usage_id, view_name):
     if not request.user.is_authenticated():
         raise PermissionDenied
 
-    instance, _ = get_module_by_usage_id(request, course_id, usage_id)
+    try:
+        course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+        usage_key = course_key.make_usage_key_from_deprecated_string(unquote_slashes(usage_id))
+    except InvalidKeyError:
+        raise Http404("Invalid location")
+    course = modulestore().get_item(usage_key)
+    instance, _ = get_module_by_usage_id(request, course_id, usage_id, course=course)
```

Comment:
`bulk_operations`. Also, `course = modulestore().get_course(course_id)`. Otherwise, you're loading an arbitrary block, and then passing it in as the `course` argument.